pkg/services/store/entity/sqlstash/sql_storage_server.go

```diff
@@ -130,6 +130,14 @@ func (s *sqlEntityServer) Init() error {
 	return nil
 }
 
+func (s *sqlEntityServer) IsHealthy(ctx context.Context, r *entity.HealthRequest) (*entity.HealthResponse, error) {
+	if err := s.Init(); err != nil {
```

Comment:
I don't thin we can use the Init() method to infer the service is healthy. Init doesn't seem idempotent ? 

I think this should include:
- Are all the dskit services running ? 
- An active check on the backend database (i.e SELECT 1 for SQL)

Reply:
ya good idea, I'll add that 👍